metrics-httpclient/src/main/java/com/codahale/metrics/httpclient/InstrumentedHttpClientConnectionManager.java

```diff
@@ -30,80 +30,48 @@
-        this(metricsRegistry, socketFactoryRegistry, -1, TimeUnit.MILLISECONDS);
-    }
-
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
-                                                   Registry<ConnectionSocketFactory> socketFactoryRegistry,
-                                                   long connTTL,
-                                                   TimeUnit connTTLTimeUnit) {
-        this(metricsRegistry, socketFactoryRegistry, null, null, SystemDefaultDnsResolver.INSTANCE, connTTL, connTTLTimeUnit, null);
-    }
-
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
-                                                   Registry<ConnectionSocketFactory> socketFactoryRegistry,
-                                                   HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection>
-                                                           connFactory,
-                                                   SchemePortResolver schemePortResolver,
-                                                   DnsResolver dnsResolver,
-                                                   long connTTL,
-                                                   TimeUnit connTTLTimeUnit,
-                                                   String name) {
-        this(metricsRegistry,
-             new DefaultHttpClientConnectionOperator(socketFactoryRegistry, schemePortResolver, dnsResolver),
-             connFactory,
-             connTTL,
-             connTTLTimeUnit,
-             name);
-    }
-
-    public InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
+    private InstrumentedHttpClientConnectionManager(MetricRegistry metricsRegistry,
                                                    HttpClientConnectionOperator httpClientConnectionOperator,
                                                    HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection>
                                                            connFactory,
                                                    long connTTL,
                                                    TimeUnit connTTLTimeUnit,
                                                    String name) {
         super(httpClientConnectionOperator, connFactory, connTTL, connTTLTimeUnit);
+        if (metricsRegistry==null) {
+            throw new IllegalArgumentException("metricsRegistry must be specified");
+        }
         this.metricsRegistry = metricsRegistry;
         this.name = name;
 
         metricsRegistry.register(name(HttpClientConnectionManager.class, name, "available-connections"),
-            (Gauge<Integer>) () -> {
-                // this acquires a lock on the connection pool; remove if contention sucks
-                return getTotalStats().getAvailable();
-            });
+                (Gauge<Integer>) () -> {
```

Comment:
The same comment about the whitespacing changes. Also, please take a look at the warnings produced by `checkstyle`.

Reply:
The whitespacing was changed by the IntelliJ (which, I think, is a good thing, as long as all users have consistently configured autoformatters). But I'll undo it, no problem.